modules/Rock.cmake

```diff
@@ -1350,6 +1368,39 @@ function(rock_gtest TARGET_NAME)
 
     rock_setup_gtest_test(${TARGET_NAME} ${GMOCK_DIR} ${GTEST_DIR})
     rock_add_test(${TARGET_NAME} "${__rock_test_parameters}")
+    if (ROCK_TEST_CXX_GCOVR_GENERATION_ENABLED)
+        rock_coverage_report_generation(${PROJECT_NAME} ${TEST_TARGET_NAME})
+    endif()
+endfunction()
+
+function(rock_coverage_report_generation TARGET_NAME TEST_TARGET)
```

Comment:
I would prefer the name `rock_cxx_coverage_report`

About the `TEST_TARGET`. It is here to add dependencies ... I think it would be best to make this explicit in the interface, and allow more than one dependency:

```
rock_cxx_coverage_report(${PROJECT_NAME} DEPENDS test-something-cxx test-somethingelse-cxx)
```

Reply:
I couldn't get the dependencies to work as stated above, either way i pushed a fix using the new ALL_TEST_TARGETS variable for now, if you have the time can you help me take a look on it? I've used this as the base for the implementation:
https://cmake.org/cmake/help/latest/prop_test/DEPENDS.html

Reply:
I couldn't get the dependencies to work as stated above, either way i pushed a fix using the new ALL_TEST_TARGETS variable for now, if you have the time can you help me take a look on it? I've used this as the base for the implementation:
https://cmake.org/cmake/help/latest/prop_test/DEPENDS.html